### ISSUE TITLE
Engine include app.import hook fails for environment specific assets

### ISSUE THREAD

**Issue by nmcclay** [ember-engines/ember-engines#280]
For example an import like this will succeed:
`target.import(target.bowerDirectory + '/spin.js/spin.js');`

But an import as specified in the [Ember-CLI Guide -> Environment specific assets](https://ember-cli.com/user-guide/#environment-specific-assets)  will fail:
```
target.import({
    development: target.bowerDirectory + '/spin.js/spin.js',
    production:  target.bowerDirectory + '/spin.js/spin.min.js',
    distribution: target.bowerDirectory + '/spin.js/spin.min.js'
});
```

Error output:
```
asset.replace is not a function
TypeError: asset.replace is not a function
    at Class.import (/Users/nmcclay/git/my-ember-project/node_modules/ember-engines/lib/engine-addon.js:241:17)
```

Clearly shows that its failing on this regex shim for non-lazyLoading engines:
https://github.com/ember-engines/ember-engines/blob/master/lib/engine-addon.js#L241

**Comment by nathanhammond** [comment]
@nmcclay You're correct. I didn't actually cover that use case in my work for getting `import` to work as cleverly as possible. You've actually traced exactly the spot where you need add an environment check and load the correct asset. Would you like to add that as a PR?

**Comment by ima007** [comment]
I actually created a PR for this here back on Nov 7, haven't verified quite yet, but feel free to take it on: https://github.com/ember-engines/ember-engines/pull/252
